nixos/modules/system/boot/plymouth.nix

```diff
@@ -167,12 +167,25 @@ in
     systemd.services.plymouth-poweroff.wantedBy = [ "poweroff.target" ];
     systemd.services.plymouth-reboot.wantedBy = [ "reboot.target" ];
     systemd.services.plymouth-read-write.wantedBy = [ "sysinit.target" ];
+    systemd.services.plymouth-switch-root-initramfs.wantedBy = [
+      "halt.target"
+      "kexec.target"
+      "poweroff.target"
+      "reboot.target"
+      "plymouth-switch-root-initramfs.service"
+    ];
+    systemd.services.plymouth-switch-root-initramfs.after = [
+      "generate-shutdown-ramfs.service"
```

Comment:
https://gitlab.freedesktop.org/plymouth/plymouth/-/blob/main/systemd-units/plymouth-switch-root-initramfs.service.in?ref_type=heads#L7

This should be the nixos equivalent, we need to have the shutdown ramfs ready before switching plymouth back into initramfs mode.